drivers/pds/api/backupTarget.go

```diff
@@ -52,14 +53,12 @@ func (backupTarget *BackupTarget) GetBackupTarget(backupTargetID string) (*pds.M
 	backupTargetClient := backupTarget.apiClient.BackupTargetsApi
 	ctx, err := pdsutils.GetContext()
 	if err != nil {
-		log.Errorf("Error in getting context for api call: %v\n", err)
-		return nil, err
+		return nil, fmt.Errorf("Error in getting context for api call: %v\n", err)
 	}
 	backupTargetModel, res, err := backupTargetClient.ApiBackupTargetsIdGet(ctx, backupTargetID).Execute()
 
 	if res.StatusCode != status.StatusOK {
-		log.Errorf("Error when calling `ApiBackupTargetsIdGet``: %v\n", err)
-		log.Errorf("Full HTTP response: %v\n", res)
+		return nil, fmt.Errorf("Error when called `ApiBackupTargetsIdGet`, Full HTTP response: %v\n", res)
```

Comment:
check for err also

Reply:
Since most of the changes are around the code improvement in existing code(i.e Error logging etc). Hence we will track the improvement as part of the following ticket https://portworx.atlassian.net/browse/PA-678.